plasmapy/atomic/nuclear.py

```diff
@@ -0,0 +1,186 @@
+    >>> (after - before).to(u.MeV)  # released energy from D + T --> alpha + n
+    <Quantity 17.589296207151556 MeV>
+
+    """
+
+    if argument == 'n' and mass_numb is None or mass_numb == 1:
+        return 0.0 * units.J
+
+    isotopic_symbol = isotope_symbol(argument, mass_numb)
+    isotopic_mass = isotope_mass(isotopic_symbol)
+    number_of_protons = atomic_number(argument)
+
+    if mass_numb is None:
+        mass_numb = mass_number(argument)
+    number_of_neutrons = mass_numb - number_of_protons
+
+    if number_of_protons == 1 and number_of_neutrons == 0:
+        binding_energy = 0.0 * units.J
+    else:
+        mass_of_nucleons = (number_of_protons * constants.m_p +
+                            number_of_neutrons * constants.m_n)
+        mass_defect = mass_of_nucleons - isotopic_mass
+        binding_energy = mass_defect * constants.c**2
+
+    return binding_energy.to(units.J)
+
+
+def nuclear_reaction_energy(reaction):
+    """Returns the released energy from a nuclear fusion reaction.
+
+    Parameters
+    ----------
+    reaction: string
+        A string representing the reaction, like "D + T --> alpha + n"
+
+    Returns
+    -------
+    energy: Quantity
+        The change in nuclear binding energy, which will be positive
+        if the reaction releases and negative if the reaction is
+        energetically unfavorable.
+
+    Raises
+    ------
+    ValueError:
+        If the input is not a valid reaction, there is insufficient
+        information to determine an isotope, or if the number of
+        nucleons is not conserved during the reaction.
+
+    TypeError:
+        If the input is not a string.
+
+    See also
+    --------
+    nuclear_binding_energy : finds the binding energy of an isotope
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> nuclear_reaction_energy("D + T --> alpha + n")
+    <Quantity 17.589296207151556 MeV>
+    >>> triple_alpha1 = 'alpha + He-4 --> Be-8'
+    >>> triple_alpha2 = 'Be-8 + alpha --> carbon-12'
+    >>> energy_triplealpha1 = nuclear_reaction_energy(triple_alpha1)
+    >>> energy_triplealpha2 = nuclear_reaction_energy(triple_alpha2)
+    >>> print(energy_triplealpha1, '\n', energy_triplealpha2)
+    -0.09183948324626812 MeV
+    7.366586766240317 MeV
+    >>> energy_triplealpha1.to(u.J)
+    <Quantity -1.471430677988809e-14 J>
+    >>> energy_triplealpha2.cgs
+    <Quantity -1.4714306779888094e-07 erg>
+    >>> nuclear_reaction_energy('alpha + alpha --> 2alpha')
+    <Quantity 0.0 MeV>
+
+    """
+
+    import re
+
+    def _get_isotopes_list(side):
+        """Parse a side of a reaction to get a list of the isotopes."""
+        pre_list = re.split(' \+ ', side)
+        isotopes_list = []
+        for item in pre_list:
+            item = item.strip()
+            try:
+                if item == 'n':
+                    symbol = 'n'
+                else:
+                    symbol = isotope_symbol(item)
```

Comment:
I'm still not sure what's worse, the weird feeling I get seeing how this has to be handled separately, or calling a neutron an isotope.

Reply:
Good catch...  Since `isotope_symbol` works with neutrons, we can shorten these four lines down to just `symbol = isotope_symbol(item)`.  I may change `symbol` to `isotope` for clarity and consistency as well.